### ISSUE TITLE
[FR] enable customizing the name of 'Step' in log_metric. Display x axis with negative value in mlflow ui

### ISSUE THREAD

**Issue by guanboch** [mlflow/mlflow#13485]
### Willingness to contribute

Yes. I would be willing to contribute this feature with guidance from the MLflow community.

### Proposal Summary

In mlflow UI, for the log_metric function, I would like to have the X axis to be a different parameter instead of Step. For example, the X axis can be the current 'SNR' metric of the iteration,  while the Y axis can be the validation loss. The value of the 'SNR' can be negative

### Motivation

> #### What is the use case for this feature?
Enable log_metric to have a customizable x axis instead of Step. The value of the x axis can be negative, and it will display correctly in the MLflow ui
> #### Why is this use case valuable to support for MLflow users in general?
During experiment, the logged metric might not only associate with the step but also associate with other important metric at that step. I would like to plot  (SNR, val_loss) in the mlflow UI. The 'SNR' metric was in the place of the Step. 
> #### Why is this use case valuable to support for your project(s) or organization?
It will add lots of flexibility to log the metrics and display them against an x-axis which is more relevant than Steps. 
> #### Why is it currently difficult to achieve this use case?
We just need to add this feature. Thank you!

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by WeichenXu123** [comment]
The proposal makes sense to me. I will raise the idea in our team meeting and see opinion from other folks  :)

**Comment by WeichenXu123** [comment]
Our TL @BenWilson2 also agrees with your idea. I assign this to you  :) Appreciate your contribution!

**Comment by msiron-entalpic** [comment]
I would also be interested in this feature!

**Comment by peacefulotter** [comment]
I am very surprised this isn't implemented. 

I wanted to plot some metrics every epoch, thus changing the x-axis label to "Epoch"
